integration-tests/relayinterface/chain_components_test.go

```diff
@@ -923,7 +937,13 @@ func (it *SolanaChainComponentsInterfaceTester[T]) buildContractReaderConfig(t T
 	pdaDataPrefix := []byte("data")
 	pdaDataPrefix = binary.LittleEndian.AppendUint64(pdaDataPrefix, idx)
 	pdaStructDataPrefix := []byte("struct_data")
-	pdaStructDataPrefix = binary.LittleEndian.AppendUint64(pdaStructDataPrefix, idx)
+	pdaStructData1Prefix := make([]byte, len(pdaStructDataPrefix))
+	copy(pdaStructData1Prefix, pdaStructDataPrefix)
+	pdaStructData1Prefix = binary.LittleEndian.AppendUint64(pdaStructData1Prefix, 1)
+	pdaStructData1Prefix = binary.LittleEndian.AppendUint64(pdaStructData1Prefix, idx)
+	pdaStructData2Prefix := make([]byte, len(pdaStructData1Prefix))
+	copy(pdaStructData2Prefix, pdaStructData1Prefix)
+	binary.LittleEndian.PutUint64(pdaStructData2Prefix[11:19], 2)
```

Comment:
nit: couldn't we just do the same thing as `pdaStructData1Prefix` here so you don't have to replace specific indexes?
```suggestion
	pdaStructData2Prefix := make([]byte, len(pdaStructDataPrefix))
	copy(pdaStructData2Prefix, pdaStructDataPrefix)
	pdaStructData2Prefix = binary.LittleEndian.AppendUint64(pdaStructData2Prefix, 2)
	pdaStructData2Prefix = binary.LittleEndian.AppendUint64(pdaStructData2Prefix, idx)
```